localstack/services/s3/utils.py

```diff
@@ -307,14 +307,8 @@ def parse_copy_source_range_header(copy_source_range: str, object_size: int) ->
 
 
 def get_full_default_bucket_location(bucket_name: BucketName) -> str:
-    if config.HOSTNAME_EXTERNAL != config.LOCALHOST:
-        host_definition = localstack_host(
-            use_hostname_external=True, custom_port=config.get_edge_port_http()
-        )
-        return f"{config.get_protocol()}://{host_definition.host_and_port()}/{bucket_name}/"
-    else:
-        host_definition = localstack_host(use_localhost_cloud=True)
-        return f"{config.get_protocol()}://{bucket_name}.s3.{host_definition.host_and_port()}/"
+    host_definition = localstack_host()
+    return f"{config.get_protocol()}://{bucket_name}.s3.{host_definition.host_and_port()}/"
```

Comment:
I see why we wouldn't default to the virtual host based routing here when the host is not the default Localstack one: we can't make sure that it will be resolvable with subdomains. I wonder if we should keep a similar logic here?

Reply:
I am sort of leaving that issue up to the user. The DNS server resolves all subdomains of `LOCALSTACK_HOST` to the LS container (where `LOCALSTACK_HOST` defaults to `localhost.localstack.cloud`). 

I agree I don't think we are ready yet to switch to domain support across the board. I think I will keep the previous behaviour in this case.

Reply:
@bentsku is there a reason this function is duplicated in the `provider.py`? https://github.com/localstack/localstack/pull/9390/files#diff-495eb5181d51cd13d47f8e42807cae85091f4f61c0b3ba27d18cbcdda14b196aR202

Reply:
It might have been extracted for the v3 provider and I must have forgot to delete it. It can be safely deleted from the provider I believe, the provider could use the one from `utils.py`. Sorry for that! 😬 

For the domain support, thanks a lot for considering it! Can't wait for our users to use our DNS server 👍